### PR TITLE
ItemGroup mappings

### PR DESCRIPTION
**Pull request by EnnuiL**
The focus was to cover the rifts caused by the new snapshot, however, I may consider revisiting item groups and the creative inventory later on if I decide that the names need further improving

This PR is best combined with PR #253

### CODE REVIEW

**Review by LemmaEOF — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]

**Comment by EnnuiL** [comment]
PR #253 mentions that `SplitStackCollector` might not be a good name; Maybe we should nab the `Gated` part of that name

File: mappings/net/minecraft/item/ItemGroup.mapping [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 	FIELD f_ncvkfasf texture Ljava/lang/String;
 	FIELD f_sfwfnelq scrollbar Z
 	FIELD f_xkfcdxhm index I
 	FIELD f_ypvwgfta renderName Z
+	METHOD <init> (ILnet/minecraft/unmapped/C_rdaqiwdt;)V
+		ARG 1 index
+		ARG 2 name
 	METHOD m_bvcnuvet isSpecial ()Z
+	METHOD m_bwauoqmb searchContains (Lnet/minecraft/unmapped/C_czxxrbcp;Lnet/minecraft/unmapped/C_sddaxwyk;)Z
+	METHOD m_cfeptysp getOrInitStacks (Lnet/minecraft/unmapped/C_czxxrbcp;Z)Lnet/minecraft/unmapped/C_voakmxnc;
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemGroup.mapping:15]: `getOrCreateStacks`?
```suggestion
	METHOD m_cfeptysp getOrCreateStacks (Lnet/minecraft/unmapped/C_czxxrbcp;Z)Lnet/minecraft/unmapped/C_voakmxnc;
```

File: mappings/net/minecraft/item/ItemGroup.mapping [hunk 2]
```diff
@@ -15,11 +23,32 @@ CLASS net/minecraft/unmapped/C_lnokcayd net/minecraft/item/ItemGroup
 		COMMENT <p>The name is rendered below the top row of item groups and above the inventory.
 	METHOD m_iwnppnlh setTexture (Ljava/lang/String;)Lnet/minecraft/unmapped/C_lnokcayd;
 		ARG 1 texture
+	METHOD m_mziozlja clearCache ()V
+	METHOD m_npmupqba provideStacks (Lnet/minecraft/unmapped/C_czxxrbcp;Lnet/minecraft/unmapped/C_lnokcayd$C_wnogvxsu;)V
+	METHOD m_nsgpxdxb getOrInitSearchStacks (Lnet/minecraft/unmapped/C_czxxrbcp;)Lnet/minecraft/unmapped/C_voakmxnc;
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemGroup.mapping:28]: `getOrCreateSearchStacks`?
```suggestion
	METHOD m_nsgpxdxb getOrCreateSearchStacks (Lnet/minecraft/unmapped/C_czxxrbcp;)Lnet/minecraft/unmapped/C_voakmxnc;
```

File: mappings/net/minecraft/item/ItemGroup.mapping [hunk 3]
```diff
@@ -15,11 +23,32 @@ CLASS net/minecraft/unmapped/C_lnokcayd net/minecraft/item/ItemGroup
...
 	METHOD m_ojgukxpb setNoScrollbar ()Lnet/minecraft/unmapped/C_lnokcayd;
 	METHOD m_oxvpyaqs getTexture ()Ljava/lang/String;
+	METHOD m_qngoqhkn getOrInitTabStacks (Lnet/minecraft/unmapped/C_czxxrbcp;)Lnet/minecraft/unmapped/C_voakmxnc;
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemGroup.mapping:31]: `getOrCreateTabStacks`?
```suggestion
	METHOD m_qngoqhkn getOrCreateTabStacks (Lnet/minecraft/unmapped/C_czxxrbcp;)Lnet/minecraft/unmapped/C_voakmxnc;
```

File: mappings/net/minecraft/item/ItemGroup.mapping [hunk 4]
```diff
@@ -15,11 +23,32 @@ CLASS net/minecraft/unmapped/C_lnokcayd net/minecraft/item/ItemGroup
...
 	METHOD m_qsfftnyy hideName ()Lnet/minecraft/unmapped/C_lnokcayd;
 		COMMENT Specifies that when this item group is selected, the name of the item group should not be rendered.
-	METHOD m_tcbipafs getTranslationKey ()Lnet/minecraft/unmapped/C_rdaqiwdt;
+	METHOD m_tcbipafs getName ()Lnet/minecraft/unmapped/C_rdaqiwdt;
 	METHOD m_tkahlerj getColumn ()I
 	METHOD m_uprbbsoi getIcon ()Lnet/minecraft/unmapped/C_sddaxwyk;
 	METHOD m_uribjksx getIndex ()I
+	CLASS C_qjyuivjw SplitStackCollector
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemGroup.mapping]:
```suggestion
	CLASS C_qjyuivjw GatedStackCollector
```

File: mappings/net/minecraft/item/ItemGroups.mapping [hunk 5]
```diff
@@ -0,0 +1,25 @@
+CLASS net/minecraft/unmapped/C_debvfuic net/minecraft/item/ItemGroups
+	FIELD f_bmtsybyi SURVIVAL_INVENTORY_GROUP Lnet/minecraft/unmapped/C_lnokcayd;
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemGroups.mapping:2]: Why is this "survival inventory"? Shouldn't it be just "inventory"?

Comment (EnnuiL) [reply mappings/net/minecraft/item/ItemGroups.mapping:2]: That's the full name of the tab on the English localization; And plus, only INVENTORY can be a bit ambiguous

File: mappings/net/minecraft/item/ItemGroup.mapping (hunk 2 above)
Comment (EnnuiL) [reply mappings/net/minecraft/item/ItemGroup.mapping:28]: The contribution doc says that `create` should be avoided for lazy initialization, so I searched for alternatives inside QM and got `init`

Comment (IotaBread) [reply mappings/net/minecraft/item/ItemGroup.mapping:28]: Oh, my bad then

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/item/LinkedItemStackHashSet.mapping [hunk 6]
```diff
@@ -0,0 +1,8 @@
+CLASS net/minecraft/unmapped/C_voakmxnc net/minecraft/item/LinkedItemStackHashSet
+	FIELD f_bbmejudb strategy Lit/unimi/dsi/fastutil/Hash$Strategy;
```
Comment (IotaBread) [inline mappings/net/minecraft/item/LinkedItemStackHashSet.mapping]:
```suggestion
	FIELD f_bbmejudb STRATEGY Lit/unimi/dsi/fastutil/Hash$Strategy;
```

Comment (EnnuiL) [reply mappings/net/minecraft/item/LinkedItemStackHashSet.mapping]: hm, i swear that it was a private final field

**Review by NoComment1105 — APPROVED** [review]

File: mappings/net/minecraft/item/LinkedItemStackHashSet.mapping (hunk 6 above)
Comment (EnnuiL) [reply mappings/net/minecraft/item/LinkedItemStackHashSet.mapping]: wait no, you're right, it's a private static final field

**Comment by EnnuiL** [comment]
I've addressed the issues and updated it for 22w43a; I think we're ready to go!

**Review by OroArmor — APPROVED** [review]

**Pull request merged**